File: java-to-go-projects/06-json-config-merger/Loader.java
package merger;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.*;
import java.nio.file.*;

/**
 * Loads JSON files from disk into JsonNode trees.
 */
public class Loader {

    private static final ObjectMapper mapper = new ObjectMapper();

    public static JsonNode loadFile(String filename) throws IOException {
        String content = Files.readString(Path.of(filename));
        return mapper.readTree(content);
    }

    public static String prettyPrint(JsonNode node) throws IOException {
        return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(node);
    }
}

File: java-to-go-projects/06-json-config-merger/rust/src/loader.rs
use serde_json::Value;
use std::fs;

/// Load a JSON file from disk and return as serde_json::Value.
pub fn load_file(filename: &str) -> Result<Value, String> {
    todo!()
}

/// Pretty-print a serde_json::Value to a formatted JSON string.
pub fn pretty_print(value: &Value) -> Result<String, String> {
    todo!()
}
